src/cmd-osbuild

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+dn=$(dirname "$0")
+# shellcheck source=src/cmdlib.sh
+. "${dn}"/cmdlib.sh
+
+# A list of supported platforms and the filename suffix of the main
+# artifact that platform produces.
+declare -A SUPPORTED_PLATFORMS=(
+    ['applehv']='raw.gz'
+    ['gcp']='tar.gz'
+    ['hyperv']='vhdx.zip'
+    ['metal4k']='raw'
+    ['metal']='raw'
+    ['qemu']='qcow2'
+    ['qemu-secex']='qcow2'
+)
+
+print_help() {
+    cat 1>&2 <<EOF
+Usage: coreos-assembler osbuild --help
+       coreos-assembler osbuild --supported-platforms
+       coreos-assembler osbuild [--build ID] qemu metal metal4k...
+
+  Build artifacts for the given platforms using OSBuild.
+EOF
+}
+
+
+# Parse the passed config JSON and extract a mandatory value
+getconfig() {
+    k=$1
+    config=$2
+    jq -re .\""$k"\" < "${config}"
+}
+# Return a configuration value, or default if not set
+getconfig_def() {
+    k=$1
+    shift
+    default=$1
+    config=$2
+    jq -re .\""$k"\"//\""${default}"\" < "${config}"
+}
+
+# Store information about the artifact into meta.json and also
+# "finalize" it, copying it into the builddir.
+postprocess_artifact() {
+    local artifact_name=$1
+    local local_filepath=$2
+    local target_filename=$3
+    local skip_compress=$4
+    if [[ ! "${skip_compress}" =~ ^(True|False)$ ]]; then
+        fatal "Must specify 'True' or 'False' for skip_compress. Provided: '${skip_compress}'"
+    fi
+    cosa meta --workdir "${workdir}" --build "${build}" --dump | python3 -c "
+import sys, json
+j = json.load(sys.stdin)
+j['images']['${artifact_name}'] = {
+    'path': '${target_filename}',
+    'sha256': '$(sha256sum_str < "${local_filepath}")',
+    'size': $(stat -c '%s' "${local_filepath}")
+}
+# backwards conditional here because of a invalid (for this code) CI check
+# https://github.com/coreos/coreos-assembler/pull/3930#issuecomment-2473714222
+if False:
+    pass
+elif ${skip_compress}:
+    j['images']['${artifact_name}']['skip-compression'] = True
```

Comment:
Heh, that's actually kinda funny.

I think another way around this is:

```
if True == ${skip_compression}:
    ...
```

Not worth the respin